Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/storage/memory/MemoryNode.java

```diff
@@ -0,0 +1,1096 @@
+         */
+        Iterator<ObjectType> inner;
+        /**
+         * Outer iterator to iterate over the map of relations.
+         */
+        Iterator<Entry<String, Collection<ObjectType>>> outer;
+
+        /**
+         * Creates a new ChildNodeIterator.
+         */
+        private ChildNodeIterator() {
+            outer = edges.entrySet().iterator();
+        }
+
+        /**
+         * Looks for a next node.
+         *
+         * @return a next node, or null if there is none
+         */
+        private ObjectType findNextNode() {
+            while (true) {
+                if ((inner == null) || !inner.hasNext()) {
+                    if (outer.hasNext()) {
+                        inner = outer.next().getValue().iterator();
+                    } else {
+                        return null;
+                    }
+                }
+                while (inner.hasNext()) {
+                    return inner.next();
+                }
+            }
+        }
+
+        /**
+         * Returns whether there is a next node retrievable.
+         */
+        @Override
+        public boolean hasNext() {
+            if (found == null) {
+                found = findNextNode();
+            }
+            return found != null;
+        }
+
+        /**
+         * Returns the next node.
+         *
+         * @throws NoSuchElementException
+         *             if there is none.
+         */
+        @Override
+        public ObjectType next() {
+            if (found == null) {
+                found = findNextNode();
+            }
+            if (found == null) {
+                throw new NoSuchElementException();
+            }
+            ObjectType n = found;
+            found = null;
+            return n;
+        }
+
+        /**
+         * The iterator does not support removing.
+         */
+        @Override
+        public void remove() {
+            throw new UnsupportedOperationException();
+        }
+
+    }
+
+    /**
+     * The pair of the first and last index in use, as return type of the
+     * {@code range()} function.
+     */
+    private class Indices {
+        /**
+         * The first index in use.
+         */
+        private final long first;
+        /**
+         * The last index in use.
+         */
+        private final long last;
+
+        /**
+         * Creates a new pair of indices.
+         *
+         * @param first
+         *            first index in use
+         * @param last
+         *            last index in use
+         */
+        public Indices(long first, long last) {
+            this.first = first;
+            this.last = last;
+        }
+    }
+
+    /**
+     * The edges leading from this node to other nodes and leaves.
+     */
+    protected final HashMap<String, Collection<ObjectType>> edges = new HashMap<>();
+
+    /**
+     * Creates an empty node.
+     */
+    public MemoryNode() {
```

Comment:
Not needed declaration of the constructor.

Reply:
The constructor is needed or you cannot create an empty object. The default constructor is not created if the class has other constructors, which is the case.